Automattic/wp-calypso
# Add the WPCOM Block Editor NUX

#### Changes proposed in this Pull Request

⚠️ I've opened this for reviews, but I definitely need to double check the test steps to make sure it actually works as expected.

* Add a custom WPCOM Block Editor NUX to the FSE plugins.
* Override the Core NUX (Welcome Guide) to trigger WPCOM's instead.
* If on WPCOM, the NUX is per-user instead of per-site. (See D36911-code)

<img width="953" alt="Screenshot 2019-12-18 at 18 54 33" src="https://user-images.githubusercontent.com/2070010/71114509-dab93200-21c7-11ea-8bb7-20bc97e08cdd.png">

#### Notes

Currently the WPCOM NUX has the exact same copy as the Core NUX.

Although, the WPCOM NUX is intentionally without images, so that it won't be confused with the Core NUX while testing.

#### Testing instructions

For the WPCOM instructions, please refer to D36911-code.

* Open the block editor and make sure the WPCOM NUX show up.
It's the first time, so it's expected to show up even if we had previously dismissed the Core NUX already.
* Dismiss the NUX.
* Reload the editor: make sure the NUX does not show up.
* Open the browser dev tools, and navigate to: Application -> Local Storage -> [site URL, e.g. localhost:8889]. Wipe the local storage for your site by clicking on the 🚫icon.

<img width="809" alt="Screenshot 2019-12-18 at 18 47 10" src="https://user-images.githubusercontent.com/2070010/71114790-7480df00-21c8-11ea-81a1-59f7f8d5284e.png">

* Reload the editor: make sure the NUX does not show up.
* From the More menu, click on Welcome Guide. The NUX will show up again.
* Without dismissing the NUX, reload the editor: make sure the NUX will show up again.

Fixes #38299 & #39320
